vpr/src/place/initial_noc_placement.cpp

```diff
@@ -275,4 +276,12 @@ void initial_noc_placement(const t_noc_opts& noc_opts, int seed) {
 
     // Run the simulated annealing optimizer for NoC routers
     noc_routers_anneal(noc_opts);
+
+    // check if there is any cycles
+    bool has_cycle = noc_routing_has_cycle();
+    if (has_cycle) {
+        VPR_FATAL_ERROR(VPR_ERROR_PLACE,
+                        "At least one cycle was found in NoC channel dependency graph. This may cause a deadlock "
+                        "when packets wait on each other in a cycle.\n");
+    }
```

Comment:
I think you also check this in check_place now ... would be a good idea to add if you don't.